### PR TITLE
10541 add txn handling mode to loan

### PR DESCRIPTION
**Pull request by DevneyHamilton**
Add field to loan `txn_handling_mode` to loan. It defaults to "automatic." If 'automatic', Madeline will treat loan as before (a loan update will prompt interest calculations, and admins can create txns). If set to read-only, interest calculation will not run for this loan's txns, and admins cannot create txns. Overall, if set to read-only, QB data for this loan will not be changed by Madeline, but Madeline will still import Qb data and it is available in the loan txn view and reports.

### CODE REVIEW

**Comment by DevneyHamilton** [comment]
**User facing changes**

In the loan details, there is a new field, Transaction Mode. It defaults to 'automatic'. The other option is 'read-only.'
Screen Shot 2020-03-24 at 11 29 33 AM
Screen Shot 2020-03-24 at 11 29 21 AM


In the the transactions view for a loan, if the loan has its transaction mode set to read-only, the user will 1) see an info bar saying 'Transactions are read-only because the loan is not active, transactions are in read-only mode for this loan, or both.' and 2) NOT see the 'Add Transaction' button. Details on this behavior with various settings are at https://docs.google.com/spreadsheets/d/1Lz9WB2453PUipVJXjm7Tum0LTzc1rvOcG1T_SZv8rr8/edit?usp=drive_web&ouid=104077519992976290870

Screen Shot 2020-03-24 at 11 55 16 AM

**Behind the scenes changes**

If a loan's txn mode is set to read-only, the interest calculator should not run on it.

**Comment by DevneyHamilton** [comment]
@emitche would you be able to address the formatting with the radio buttons in the loan form? I think you would be able to do it faster than I can.

**Review by emitche — APPROVED** [review]
The code looks good.

File: app/assets/stylesheets/admin/forms/_forms.scss [hunk 1]
```diff
@@ -196,6 +196,10 @@ label.light {
   &.form-with-documentation {
     .form-control {
       width: 90%;
+
+      &.radio_buttons {
```
[the remark this answers is not in the document]
Comment (emitche) [reply app/assets/stylesheets/admin/forms/_forms.scss:200]: This is an automatically generated CSS class.

**Comment by emitche** [comment]
Form improvements. I think it could use further improvement related to the documentation circles, but this will do. I also added the option for admin-generated documentation.

image

**Comment by DevneyHamilton** [comment]
> Form improvements. I think it could use further improvement related to the documentation circles, but this will do. I also added the option for admin-generated documentation.
> 
> image

Thank you so much @emitche!!

**Comment by emitche** [comment]
> > Form improvements. I think it could use further improvement related to the documentation circles, but this will do. I also added the option for admin-generated documentation.
> > image
> 
> Thank you so much @emitche!!

Thank you, @DevneyHamilton!

**Comment by DevneyHamilton** [comment]
The hound error on this won't resolve because I rebased & github can't tell its fixed.

**Comment by DevneyHamilton** [comment]
Oh, the CI builds also passed but are not showing updated here for some reason (click on 'Details' to see they passed).

**Comment by emitche** [comment]
Thanks, @DevneyHamilton!

**Comment by emitche** [comment]
> Oh, the CI builds also passed but are not showing updated here for some reason (click on 'Details' to see they passed).

This happened on another pull request recently. Yes, the tests did pass. Thank you!

**Pull request merged**